components/plugin_marketplace/marketplace_usage_message.tsx

```diff
@@ -0,0 +1,137 @@
+    const preference = preferences[key];
+    const hiddenDueToPreference = preference?.value === 'true';
+
+    const dispatch = useDispatch();
+    const setHidePreference = (value: boolean) => {
+        const preference: PreferenceType = {
+            category: HIDE_USAGE_MESSAGE_PREFERENCE_CATEGORY,
+            name: HIDE_USAGE_MESSAGE_PREFERENCE_NAME,
+            user_id: userId,
+            value: value.toString(),
+        };
+        dispatch(savePreferences(userId, [preference]));
+    };
+
+    return [hiddenDueToPreference, setHidePreference];
+};
+
+export default function MarketplaceUsageMessage() {
+    const isCloudFreeEnabled = useSelector(cloudFreeEnabled);
+    const integrationsUsage = useSelector(getIntegrationsUsage);
+    const limits = useSelector(getCloudLimits);
+    const [hiddenDueToPreferences, setHidePreference] = usePreferenceCheck();
+    const intl = useIntl();
+
+    if (hiddenDueToPreferences || !isCloudFreeEnabled || !integrationsUsage?.enabledLoaded || !limits.integrations?.enabled) {
+        return (
+            <button
+                className='btn btn-primary'
+                onClick={() => setHidePreference(false)}
+            >
+                {'DEBUG reset preference'}
+            </button>
+        );
+    }
+
+    let limitMessage: React.ReactNode;
+    if (integrationsUsage.enabled < limits.integrations.enabled) {
+        limitMessage = (
+            <FormattedMessage
+                id={'marketplace_modal.usage_message.below_limit'}
+                defaultMessage={'Your workspace is limited to {limit} enabled integrations.'}
+                values={{
+                    limit: limits.integrations.enabled,
+                }}
+            />
+        );
+    } else {
+        limitMessage = (
+            <FormattedMessage
+                id={'marketplace_modal.usage_message.reached_limit'}
+                defaultMessage={'Integrations limits reached. Upgrade to install unlimited integrations.'}
+            />
+        );
+    }
+
+    const upgradeInstructions = (
+        <p>
+            <FormattedMarkdownMessage
+                id={'marketplace_modal.usage_message.upgrade_instructions'}
+                defaultMessage={'To install unlimited integrations, upgrade to one of our paid plans. [Review our plan options and pricing]({url}).'}
+                values={{
+                    url: 'https://mattermost.com/pricing',
+                }}
+            />
+        </p>
+    );
+
+    const upgradeButtons = (
+        <div className='upgrade-buttons'>
+            <CloudStartTrialButton
+                message={'Try free for 30 days'}
+                telemetryId={'start_cloud_trial_from_marketplace'}
+                onClick={() => setHidePreference(true)}
+                extraClass={'btn btn-primary start-cloud-trial-btn'}
+            />
+            <a
+                href='https://mattermost.com/pricing'
```

Comment:
The designs say `Opens the plan page in a new tab`, which makes it sounds like it should go to this page, but I think what is meant is that a new pricing modal that recently got added to the app should be opened. @annelieseklein , can you confirm that what we want to do is open the pricing modal? And if that is the case, as this marketplace is in a modal, should we close the marketplace modal, or do a modal on top of a modal?

If what we intend is to open that modal, you can do it like this: https://github.com/mattermost/mattermost-webapp/blob/0e09cf02025056639be8f8c683b06f1e28444038/components/admin_console/billing/billing_subscriptions/limits.tsx#L22-L42

Reply:
> And if that is the case, as this marketplace is in a modal, should we close the marketplace modal, or do a modal on top of a modal?

@neallred The marketplace is technically a modal, but it is a full-screen modal, so it's basically its own page. I actually want to make it its own route/page, so you can navigate back to the marketplace from the system console, rather than manually navigating back to the marketplace once you're back in the Channels UI.

Reply:
Yes it is the modal. Because it is a modal over a large page modal I am not too worried about it.

Reply:
@neallred @annelieseklein The styling is a bit off when the modals are stacked. Notice the title of the modal is cut off. Also, if I have my bookmarks bar open, more content gets cut off.

<img width='500px' src='https://user-images.githubusercontent.com/6913320/172255700-f3e9e4cf-f69d-43b7-9606-bb6098f052d6.png'/>

<img width='500px' src='https://user-images.githubusercontent.com/6913320/172255774-419d3f6f-0d3b-416e-97c8-9476dac6a2c8.png'/>

Reply:
@mickmister sorry just seeing this....can we make the height of the modal better adjust in web? Not sure what to do about the technical issue of ESC....what are the options?